app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_people = []
+    for person_dict in people:
+        person = Person(name=person_dict["name"], age=person_dict["age"])
+        person_people.append(person)
+
+    for person in people:
+        person_instance = Person.people.get(person["name"])
+        if person_instance:
+            if person.get("wife"):
+                wife_name = person["wife"]
+                person_instance.wife = Person.people.get(wife_name)
+            elif person.get("husband"):
+                husband_name = person["husband"]
+                person_instance.husband = Person.people.get(husband_name)
```

Comment:
Assigning `wife` or `husband` attributes directly without checking if they already exist might overwrite existing relationships. It's important to add checks or handle existing relationships appropriately to prevent data loss or corruption.